src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -0,0 +1,58 @@
+package lotto.domain;
+
+import java.util.Objects;
+import lotto.exception.InvalidLottoNumberRangeException;
+
+public class LottoNumber implements Comparable<LottoNumber> {
+
+  public static final int MIN = 1;
+  public static final int MAX = 45;
+
+  private final int number;
+  private static final LottoNumber[] lottoNumberCache = new LottoNumber[MAX];
+
+  public LottoNumber(int number) {
+    validateNumber(number);
+    this.number = number;
+  }
+
+  public static LottoNumber valueOf(int number) {
+    validateNumber(number);
+    if (lottoNumberCache[number - 1] != null) {
+      return lottoNumberCache[number - 1];
+    }
+    return lottoNumberCache[number - 1] = new LottoNumber(number);
```

Comment:
```suggestion
    validateNumber(number);
    if (lottoNumberCache[number - 1] == null) {
      lottoNumberCache[number - 1] = new LottoNumber(number);
    }
    return lottoNumberCache[number - 1];
```
이렇게 표현하는게 더 자연스러울 것 같아요!

Reply:
오 그렇네요!!! 다음단계 진행하기전에 반영해놔야겠습니다 감사합니다!!